### PR TITLE
Differentiate between "lock taken" and "error while trying to take the lock"

### PR DESCRIPTION
**Pull request by Jongy**
## Description
Errors while trying to take the lock are common (well, now it's the first "privileged" thing we're trying to do).

Running as a container w/o `--privileged` currently yields:
```
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/app/gprofiler/utils.py", line 362, in _switch_and_run
    if not is_same_ns(target_pid, nstype):
  File "/app/gprofiler/utils.py", line 291, in is_same_ns
    return os.stat(f"/proc/self/ns/{nstype}").st_ino == os.stat(f"/proc/{pid}/ns/{nstype}").st_ino
PermissionError: [Errno 13] Permission denied: '/proc/1/ns/net'
Could not acquire gProfiler's lock. Is it already running?
```

The exception is not very clear, and the message "is it already running" doesn't help.

After this PR:
```
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/app/gprofiler/utils.py", line 355, in _switch_and_run
    if not is_same_ns(target_pid, nstype):
  File "/app/gprofiler/utils.py", line 317, in is_same_ns
    return os.stat(f"/proc/self/ns/{nstype}").st_ino == os.stat(f"/proc/{pid}/ns/{nstype}").st_ino
PermissionError: [Errno 13] Permission denied: '/proc/1/ns/net'
Could not acquire gProfiler's lock due to an error. Are you running gProfiler in privileged mode?
```

Better.

## Motivation and Context
Better UX on errors

## How Has This Been Tested?
Tested w/o `--privileged` and verified the exception is as I showed above.
Tested w/ `--privileged` and ran 2 profilers - I get the old `Could not acquire gProfiler's lock. Is it already running?` correctly.

## Checklist:
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have updated the relevant documentation.
- [ ] I have added tests for new logic.

### CODE REVIEW

**Comment by michelhe** [comment]
@Jongy  ❤️ rust now ?

Also, tests fail

File: gprofiler/utils.py [hunk 1]
```diff
@@ -385,7 +386,7 @@ def grab_gprofiler_mutex() -> bool:
     """
     GPROFILER_LOCK = "\x00gprofiler_lock"
 
-    def _take_lock() -> Optional[socket.socket]:
+    def _take_lock() -> Tuple[bool, Optional[socket.socket]]:  # like Rust's Result :(
```
Comment (michelhe) [inline gprofiler/utils.py:389]: You mean, `Result<Option<T>>`?

File: gprofiler/utils.py [hunk 2]
```diff
@@ -395,20 +396,28 @@ def _take_lock() -> Optional[socket.socket]:
...
 
-    s = run_in_ns(["net"], _take_lock)
-
-    global gprofiler_mutex
-    # hold the reference so lock remains taken
-    gprofiler_mutex = s
+    res = run_in_ns(["net"], _take_lock)
+    if res is None:
+        # exception in run_in_ns
+        print("Could not acquire gProfiler's lock due to an error. Are you running gProfiler in privileged mode?")
```
Comment (michelhe) [inline gprofiler/utils.py:409]: stderror?

File: gprofiler/utils.py [hunk 3]
```diff
@@ -395,20 +396,28 @@ def _take_lock() -> Optional[socket.socket]:
...
+    elif res[0]:
+        assert res[1] is not None
 
-    return s is not None
+        global gprofiler_mutex
+        # hold the reference so lock remains taken
+        gprofiler_mutex = res[1]
+        return True
+    else:
+        print("Could not acquire gProfiler's lock. Is it already running?", file=sys.stderr)
```
Comment (michelhe) [inline gprofiler/utils.py:419]: here too

Comment (michelhe) [reply gprofiler/utils.py:419]: We can be more elaborate here and write which PID holds the lock.

Comment (Jongy) [reply gprofiler/utils.py:419]: Already `sys.stderr`

File: gprofiler/utils.py (hunk 1 above)
Comment (Jongy) [reply gprofiler/utils.py:389]: Mhm

File: gprofiler/utils.py (hunk 3 above)
Comment (Jongy) [reply gprofiler/utils.py:422]: > We can be more elaborate here and write which PID holds the lock.

Hmm, it's kinda complicated to run that. I'll drop in better comment

**Review by michelhe — APPROVED** [review]
Wait for tests 😈

**Pull request merged**